src/pmem.c

```diff
@@ -49,3 +56,24 @@ void pmemThresholdInit(void)
             serverAssert(NULL);
     }
 }
+
+void adjustPmemThresholdCycle(void) {
+    if (server.memory_alloc_policy == MEM_POLICY_RATIO) {
+        run_with_period(100) {
+            //revert logic to avoid division by zero
+            float setting_state = server.dram_pmem_ratio.pmem_val/server.dram_pmem_ratio.dram_val;
+            float curent_state = zmalloc_used_pmem_memory()/zmalloc_used_memory();
```

Comment:
current

Reply:
float curent_state = (float)zmalloc_used_pmem_memory()/zmalloc_used_memory();
to get float as a result.